src/main/java/com/amazonaws/services/dynamodbv2/AmazonDynamoDBLockClient.java

```diff
@@ -497,14 +497,47 @@ public LockItem acquireLock(final AcquireLockOptions options) throws LockNotGran
                     item.put(RECORD_VERSION_NUMBER, AttributeValue.builder().s(String.valueOf(recordVersionNumber)).build());
                     sortKeyName.ifPresent(sortKeyName -> item.put(sortKeyName, AttributeValue.builder().s(sortKey.get()).build()));
                     newLockData.ifPresent(byteBuffer -> item.put(DATA, AttributeValue.builder().b(SdkBytes.fromByteBuffer(byteBuffer)).build()));
+                    Optional<AtomicLong> lastTouchedAt = options.getClockSkewUpperBound()
+                      .map(unused -> new AtomicLong(System.currentTimeMillis()));
+                    lastTouchedAt.ifPresent(value -> item.put(LAST_TOUCHED_AT, AttributeValue.builder().s(String.valueOf(value.get())).build()));
+
+                    if (options.shouldSkipBlockingWait() && existingLock.isPresent() && !existingLock.get().isExpired()) {
+                        LockItem lockItem = existingLock.get();
+                        if (options.getClockSkewUpperBound().isPresent()) {
+                            // If we have provided an upper clock skew error bound then we rely on lastTouchedAt to know
+                            // whether the lock is expired without having to block.
+                            Long clockSkewUpperBound = options.getClockSkewUpperBound().get();
+                            if (lockItem.getLastTouchedAt().isPresent()) {
+                                long currentTime = System.currentTimeMillis();
+                                // If the lock hasn't been touched since the lease duration plus error bound then it's ours!
+                                if (currentTime > lockItem.getLastTouchedAt().get().get() + lockItem.getLeaseDuration() + clockSkewUpperBound) {
```

Comment:
Implementation:

```
    /**
     * Returns whether or not the lock is expired, based on if the lock hasn't been touched since the lease duration plus cloud skew error threshold
     *
     * @param clockSkewUpperBound the amount of time to add to the lease duration to account for clock skew precision errors
     * @return True if the lock is expired, false otherwise
     */
    public boolean isExpired(Long clockSkewUpperBound) {
        if (this.isReleased) {
            return true;
        }

        if (this.lastTouchedAt.isPresent()) {
            long currentTime = System.currentTimeMillis();
            // If the lock hasn't been touched since the lease duration plus error bound then it's ours!
            return currentTime > this.lastTouchedAt.get().get() + this.leaseDuration.get() + clockSkewUpperBound;
        }

        return false;
    }
```